service/history/task/cross_cluster_task.go

```diff
@@ -470,6 +486,88 @@ func (t *crossClusterSourceTask) GetCrossClusterRequest() (request *types.CrossC
 	return request, nil
 }
 
+func (t *crossClusterSourceTask) VerifyTaskFailoverVersion(
```

Comment:
nit: since this util function specifically checks for last write version, maybe rename to something like `verifyLastWriteVersion`?